pkg/imageverifier/image_verifier.go

```diff
@@ -0,0 +1,29 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package imageverifier
+
+import "context"
+
+type ImageVerifier interface {
+	Enabled(ctx context.Context) bool
+	VerifyImage(ctx context.Context, name string, digest string) Judgement
```

Comment:
I like the simplicity but we try to use the full descriptor object over a raw digests in most interfaces. It avoids the inevitable case when media-type/size/annotations want to be used by implementations.